src/main/java/wtf/meier/tariff/interpreter/model/tariff/calculator/TimeBasedTariffCalculator.kt

```diff
@@ -70,25 +70,27 @@ class TimeBasedTariffCalculator(
         return bill.toReceipt()
     }
 
+
     fun firstIntersectingSlot(list: List<TimeBasedTariff.TimeSlot>, at: ZonedDateTime): TimeBasedTariff.TimeSlot? {
-        return list.maxByOrNull { slot ->
+        // Check all slots and choose the one with the closest starting date
+        return list.minByOrNull { slot ->
             val (day, hour, minute) = slot.from
 
-            val adjuster = TemporalAdjusters.previousOrSame(day)
-            val slotStart = at.with(adjuster)
+            // Put the slot start into the same week and timezone of the requested date
+            var adjuster = TemporalAdjusters.previousOrSame(day)
+            var slotStart = at.with(adjuster)
                     .withHour(hour)
                     .withMinute(minute)
 
-            var duration = Duration.between(at, slotStart)
-
-            // Disregard slots starting in the future
+            // Adjust the slot if it happen to be later that same day
             if (slotStart > at) {
```

Comment:
shouldn't we call it `candidate` instead of `at`?

Reply:
🤷 I didn't came up with a better idea since I can't have these external and internal names as in Swift 😁 
`func firstIntersectingSlot(for date: Date, in slots: [TimeSlot]) -> Slot?`